qualtran/_infra/data_types.py

```diff
@@ -810,8 +812,130 @@ def assert_valid_classical_val_array(
             raise ValueError(f"Too-large classical values encountered in {debug_str}")
 
 
+@attrs.frozen
+class QGF(QDType):
+    r"""Galois Field type to represent elements of a finite field.
+
+    A Finite Field or Galois Field is a field that contains finite number of elements. The order
+    of a finite field is the number of elements in the field, which is either a prime number or
+    a prime power. For every prime number $p$ and every positive integer $m$ there are fields of
+    order $p^{m}0$, all of which are isomorphic. When m=1, the finite field of order p can be
+    constructed via integers modulo p.
+
+    Elements of a Galois Field $GF(p^m)$ may be conveniently viewed as polynomials
+    $a_{0} + a_{1}x + ... + a_{m−1}x_{m−1}$, where $a_0, a_1, ..., a_{m−1} \in F(p)$.
+    $GF(p^m)$ addition is defined as the component-wise (polynomial) addition over F(p) and
+    multiplication is defined as polynomial multiplication modulo an irreducible polynomial of
+    degree $m$. The selection of the specific irreducible polynomial affects the representation
+    of the given field, but all fields of a fixed size are isomorphic.
+
+    The data type uses the [Galois library](https://mhostetter.github.io/galois/latest/) to
+    perform arithmetic over Galois Fields. By default, the Conway polynomial $C_{p, m}$ is used
+    as the irreducible polynomial.
+
+    Attributes:
+        characteristic: The characteristic $p$ of the field $GF(p^{m})$.
+            The characteristic must be prime.
+        degree: The degree $m$ of the field $GF(p^{m})$. The degree must be a positive integer.
+
+    References
+        [Finite Field](https://en.wikipedia.org/wiki/Finite_field)
```

Comment:
how does this get rendered in the api docs? Usually we separate individual references with a blank newline; but I admit that that is primarily for the notebook-based doc system

Reply:
added blank newline between references.